extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreatorTest.java

```diff
@@ -0,0 +1,620 @@
+ *
+ * <p>
+ * We couldn't use mocks for these tests because the streaming URLs needs to fetched and the IP
+ * address used to get these URLs is required (used as a param in the URLs; without it, video
+ * servers return 403/Forbidden HTTP response code).
+ * </p>
+ *
+ * <p>
+ * So the real downloader will be used everytime on this test class.
+ * </p>
+ */
+class YoutubeDashManifestCreatorTest {
+    // Setting a higher number may let Google video servers return a lot of 403s
+    private static final int MAXIMUM_NUMBER_OF_STREAMS_TO_TEST = 3;
+
+    public static class testGenerationOfOtfAndProgressiveManifests {
+        private static final String url = "https://www.youtube.com/watch?v=DJ8GQUNUXGM";
+        private static YoutubeStreamExtractor extractor;
+
+        @BeforeAll
+        public static void setUp() throws Exception {
+            YoutubeParsingHelper.resetClientVersionAndKey();
+            YoutubeParsingHelper.setNumberGenerator(new Random(1));
+            NewPipe.init(DownloaderTestImpl.getInstance());
+            extractor = (YoutubeStreamExtractor) YouTube.getStreamExtractor(url);
+            extractor.fetchPage();
+        }
+
+        @Test
+        void testOtfStreamsANewEraOfOpen() throws Exception {
+            testStreams(DeliveryMethod.DASH,
+                    extractor.getVideoOnlyStreams());
+            testStreams(DeliveryMethod.DASH,
+                    extractor.getAudioStreams());
+            // This should not happen because there are no video stream with audio which use the
+            // DASH delivery method (YouTube OTF stream type)
+            try {
+                testStreams(DeliveryMethod.DASH,
+                        extractor.getVideoStreams());
+            } catch (final Exception e) {
+                assertEquals(YoutubeDashManifestCreator.YoutubeDashManifestCreationException.class,
+                        e.getClass(), "The exception thrown was not the one excepted: "
+                                + e.getClass().getName()
+                                + "was thrown instead of YoutubeDashManifestCreationException");
+            }
+        }
+
+        @Test
+        void testProgressiveStreamsANewEraOfOpen() throws Exception {
+            testStreams(DeliveryMethod.PROGRESSIVE_HTTP,
+                    extractor.getVideoOnlyStreams());
+            testStreams(DeliveryMethod.PROGRESSIVE_HTTP,
+                    extractor.getAudioStreams());
+            try {
+                testStreams(DeliveryMethod.PROGRESSIVE_HTTP,
+                        extractor.getVideoStreams());
+            } catch (final Exception e) {
+                assertEquals(YoutubeDashManifestCreator.YoutubeDashManifestCreationException.class,
+                        e.getClass(), "The exception thrown was not the one excepted: "
+                                + e.getClass().getName()
+                                + "was thrown instead of YoutubeDashManifestCreationException");
+            }
```

Comment:
I would use [assertThrows](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertThrows-java.lang.Class-org.junit.jupiter.api.function.Executable-) here.

Reply:
Done (this change cannot be applied to the similar assertion as the exception should be never thrown because no YouTube DASH stream with video and audio exists).